lighty-modules/lighty-netconf-sb/src/main/java/io/lighty/modules/southbound/netconf/impl/util/NetconfUtils.java

```diff
@@ -177,4 +180,13 @@ public static NormalizedNode getUnLockContent(final QName targetDatastore) {
                 .withChild(getTargetNode(targetDatastore)).build();
     }
 
+    public static Optional<Absolute> getAbsolutePathToNNode(final YangInstanceIdentifier yangInstanceIdentifier,
+            final NormalizedNode normalizedNode) {
+        final List<QName> qnamePath = yangInstanceIdentifier.getPathArguments().stream()
+                .filter(pa -> !(pa instanceof YangInstanceIdentifier.AugmentationIdentifier)
+                        && !(pa instanceof YangInstanceIdentifier.NodeIdentifierWithPredicates))
+                .takeWhile(pa -> pa.getNodeType().equals(normalizedNode.getIdentifier().getNodeType()))
```

Comment:
What is takeWhile used for?

Reply:
This query collect QNames until is condition in .takeWhile method met (also is not collection QNames for Augmentation and Predicates). I have added inline description to code for case if it will be needed.